openapi/schemas/processes-dru/cwl.yaml

```diff
@@ -0,0 +1,1788 @@
+$schema: "http://json-schema.org/draft-07/schema#"
+$author: "Francis Charette-Migneault <francis.charette-migneault@crim.ca>"
+$id: "https://raw.githubusercontent.com/opengeospatial/ogcapi-processes/master/openapi/schemas/processes-dru/cwl.yaml"
+
+$definitions:
+  DeployCWL:
+    oneOf:
+      - $ref: "#/$definitions/CWL"
+      - $ref: "#/$definitions/DeployOGCAppPackageCWL"
+  DeployOGCAppPackageCWL:
+    type: object
+    title: DeployOGCAppPackage
+    description: Deployment using standard OGC Application Package definition.
+    properties:
+      processDescription:
+        $ref: "https://raw.githubusercontent.com/opengeospatial/ogcapi-processes/master/openapi/schemas/processes-core/process.yaml"
+      immediateDeployment:
+        type: boolean
+        title: immediateDeployment
+        default: true
+      deploymentProfileName:
+        type: string
+        format: uri
+        enum:
+        # if the CWL has a DockerRequirement and is a 'class: CommandLineTool'
+        - "http://www.opengis.net/profiles/eoc/dockerizedApplication"
+        # when the CWL is a 'class: Workflow'
+        - "http://www.opengis.net/profiles/eoc/workflow"
+      executionUnit:
+        oneOf:
+          - $ref: "#/$definitions/CWLExecutionUnit"
+          - type: array
+            items:
+              $ref: "#/$definitions/CWLExecutionUnit"
+    required:
+      - executionUnit
+      - processDescription
+    additionalProperties: false
+  CWLExecutionUnit:
+    description: CWL definition use to deploy a process.
+    type: object
+    properties:
+      unit:
+        $ref: '#/$definitions/CWL'
+  CWL:
+    oneOf:
+      - $ref: '#/$definitions/CWLAtomic'
+      - $ref: '#/$definitions/CWLGraph'
+      - $ref: '#/$definitions/CWLWorkflow'
+  CWLKeywordList:
+    type: array
+    title: KeywordList
+    description: Keywords applied to the process for search and categorization purposes.
+    default: []
+    items:
+      type: string
+      title: keyword
+      minLength: 1
+  CWLIdentifier:
+    anyOf:
+    - type: string
+      title: UUID
+      description: Unique identifier.
+      format: uuid
+      pattern: '^[a-f0-9]{8}(?:-?[a-f0-9]{4}){3}-?[a-f0-9]{12}$'
+    - type: string
+      title: ID
+      description: Generic identifier.
+      pattern: '^[#.]?[A-Za-zA-Za-z0-9]+(?:[-_.][A-Za-zA-Za-z0-9]+)*$'
```

Comment:
I'm confused by this `A-Za-zA-Za-z` repetition. Is that a search and replace artifact, or does that accomplish something?

Reply:
No, it is on purpose. For lowercase/uppercase match. 
My original regexes were in code using the case-insensitive flag, but it is not enabled by default in pure JSON schema.